src/index.js

```diff
@@ -65,7 +47,10 @@ const addToMailchimp = (email, fields) => {
   // generate Mailchimp endpoint for jsonp request
   // note, we change `/post` to `/post-json`
   // otherwise, Mailchomp returns an error
-  let {endpoint} = getPluginOptions()
+  let {endpoint} = window.__GATSBY_PLUGIN_MAILCHIMP__ || {};
+  if (!typeof endpoint === 'string') {
```

Comment:
Just a quick note while scanning this: this is probably not what you want to do (use `!==`) :P